imagelab/ImgProvider.java

```diff
@@ -397,18 +400,20 @@ void save() {
      * values of each row are used as the velocities
      * of the Red, Green and Blue notes respectively.
      */
-    public void play() {
+      public void play() {
+        if(playThread!=null) playThread.stop();
+        playThread = new Thread(() -> {
         short[][] red = getRed();     // Red plane
         short[][] green = getGreen(); // Green plane
         short[][] blue = getBlue();   // Blue plane
-        short[][] bw = getBWImage();  // Black & white image
+        short[][] bw = getRed();  // Black & white image
```

Comment:
Why was this change necessary?  Why should "bw" contain "red" information?

Reply:
Without this change, the image that was rendered during synchronization was in black and white since the toBW method converts the pix array from color to gray scale. I work on a fix to avoid this change.